ui/summary_test.go

```diff
@@ -164,3 +166,125 @@ func createTestMetrics() map[string]*stats.Metric {
 
 	return metrics
 }
+
+func TestSummarizeMetricsJSON(t *testing.T) {
+	metrics := createTestMetrics()
+	expected := `[{
+    "name": "",
+    "path": "",
+    "id": "d41d8cd98f00b204e9800998ecf8427e",
+    "groups": {
+        "child": {
+        "name": "child",
+        "path": "::child",
+        "id": "f41cbb53a398ec1c9fb3d33e20c9b040",
+        "groups": {},
+        "checks": {
+            "check1": {
+                "name": "check1",
+                "path": "::child::check1",
+                "id": "6289a7a06253a1c3f6137dfb25695563",
+                "passes": 5,
+                "fails": 10
+                }
+            }
+        }
+    },
+    "checks": {}
+},
+{
+    "checks": {
+        "extra": [
+            "✓ 3",
+            "✗ 0"
+        ],
+        "value": 0
+    },
+    "http_reqs": {
+        "count": 3,
+        "rate": 3
+    },
+    "my_trend": {
+        "avg": 15,
+        "max": 20,
+        "med": 15,
+        "min": 10,
+        "p(90)": 19,
+        "p(95)": 19.5
+    },
+    "vus": {
+        "extra": [
+            "min=1",
+            "max=1"
+        ],
+        "value": 1
+    }
+}
+]
```

Comment:
This hasn't been discussed but from ... easiness of parsing later I would really prefer if this is not an array but a json object with the rootGroup being in a key `root_group`/`rootGroup` and the metrics being in a key `metrics`.

Reply:
FIxed